src/index.ts

```diff
@@ -89,9 +89,9 @@ export async function findManyCursorConnection<
 
   return {
     edges: records.map((record) => ({
-      cursor: encodeCursor(record, options),
       ...options.recordToEdge(record),
-    })),
+      cursor: encodeCursor(record, options),
+    }) as CustomEdge),
```

Comment:
I couldn't do this without type assertion. I did some research an I think this is a limitation of TS: https://github.com/microsoft/TypeScript/issues/35858#issuecomment-573909154